subsys/net/lib/coap/coap.c

```diff
@@ -266,16 +349,17 @@ int coap_packet_append_option(struct coap_packet *cpkt, uint16_t code,
 	}
 
 	if (code < cpkt->delta) {
-		NET_ERR("Options should be in ascending order");
-		return -EINVAL;
+		// This should be avoided. But it can be useful in some cases.
+		NET_INFO("Option is not added in ascending order");
```

Comment:
Since out-of-order option processing is handled just fine, I'm not even sure if we need a log like this? Or at least make it a `DBG` level, IMO it doesn't need to be visiable at the default level (`INFO`).